scripts/solve_network.py

```diff
@@ -189,44 +189,50 @@ def add_chp_constraints(n):
 def add_pipe_retrofit_constraint(n):
     """Add constraint for retrofitting existing CH4 pipelines to H2 pipelines."""
 
-    gas_pipes_i = n.links[n.links.carrier=="gas pipeline"].index
-    h2_retrofitted_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].index
+    gas_pipes_i = n.links[n.links.carrier=="gas pipeline"].query("p_nom_extendable").index
+    h2_retrofitted_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].query("p_nom_extendable").index
+    h2_retrofitted_fixed_i = n.links[n.links.carrier=='H2 pipeline retrofitted'].index.difference(h2_retrofitted_i)
 
     if h2_retrofitted_i.empty or gas_pipes_i.empty: return
 
     link_p_nom = get_var(n, "Link", "p_nom")
 
-    pipe_capacity = n.links.loc[gas_pipes_i, 'p_nom']
-
     CH4_per_H2 = 1 / n.config["sector"]["H2_retrofit_capacity_per_CH4"]
-
     fr = "H2 pipeline retrofitted"
     to = "gas pipeline"
+
+    pipe_capacity = n.links.loc[gas_pipes_i, 'p_nom'].rename(index=lambda x: x.split("-2")[0])
```

Comment:
`lambda x: x.split("-2")[0]` is used quite a lot. Could we make this an own helper function? E.g.

```py
def basename(x):
     return x.split("-2")[0]
```

and then 

```
.rename(index=basename)
```